src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -377,3 +379,363 @@ class Config:
             'notifier.pwenc_encrypt', peersecret)
 
         return data
+
+
+class iSCSITargetExtentService(CRUDService):
+    class Config:
+        namespace = 'iscsi.extent'
+        datastore = 'services.iscsitargetextent'
+        datastore_prefix = 'iscsi_target_extent_'
+        datastore_extend = 'iscsi.extent.extend'
+
+    @accepts(Dict(
+        'iscsi_extent_create',
+        Str('name'),
+        Str('type'),
+        Str('disk'),
+        Str('serial', default=None),
+        Str('path'),
+        Int('filesize', default=0),
+        Int('blocksize', enum=[512, 1024, 2048, 4096], default=512),
+        Bool('pblocksize'),
+        Int('avail_threshold'),
+        Str('comment'),
+        Str('naa'),
```

Comment:
That seems to be generated automatically, should not be part of the API if I understand correctly.